server/lib/util.js

```diff
@@ -264,3 +275,69 @@ export function getCSPNonce(res : ExpressResponse) : string {
 
     return nonce;
 }
+
+/**
+ * Handler for client error, related to params and validation process
+ *
+ * @param {ErrorArgument} param0 the needed information to track and handle the errors
+ * @returns the response error
+ */
+export function clientErrorHandler({ req, res, err, logger } : ErrorArgument) : void {
+    logger.warn(req, EVENT.VALIDATION, { err: err.stack ? err.stack : err.toString() });
+    return clientErrorResponse(res, err.message);
+}
+
+/**
+ * Handler for server error and communication with the parent window.
+ * Return a HTML including the JS SDK allowing the communication with the parent SDK.
+ *
+ * @param {ErrorArgumentNonMaybeValue} param0 the needed information to track and handle the errors
+ * @returns an HTML page with access to the parent window.xprops
+ */
+export function htmlErrorHandler({ res, err, meta } : ErrorArgument) : void {
+    if (typeof meta?.getSDKLoader !== 'function') {
+        throw new TypeError('Missing "getSDKLoader" function in meta');
+    }
+    const cspNonce = getCSPNonce(res);
+    const pageHTML = `<!DOCTYPE html>
+        <head></head>
+        <body data-nonce="${ cspNonce }">
+            ${ meta.getSDKLoader({ nonce: cspNonce }) }
+            <script nonce="${ cspNonce }">
+                window?.xprops?.onError("${ err.message }");
+            </script>
+        </body>`;
+
+    return htmlResponse(res, pageHTML);
+}
+
+/**
+ * Handler for server error. If the `meta.getSDKLoader` function is not available
+ * the response will be a text and not an HTML page
+ *
+ * @param {ErrorArgument} param0 the needed information to track and handle the errors
+ * @returns the response error
+ */
+export function serverErrorHandler ({ req, res, err, logger, meta } : ErrorArgument) : void {
```

Comment:
I'd like to suggest renaming this function to `errorCodeHandler()`, assuming that's a pattern we want to extend. We do have `req.query.sdkMeta` available to us from the `req` param so I'd like to avoid having to pass `meta` though. Instead, the error handler could try unpacking and validating it and then calling onError() when it's valid.

```suggestion
export function errorCodeHandler ({ req, res, err, logger } : ErrorArgument) : void {
```